### PR TITLE
auto parse attribute type in ir

### PR DESCRIPTION
**Pull request by typhoonzero**
Part work of https://github.com/sql-machine-learning/sqlflow/issues/805

### CODE REVIEW

File: sql/ir_generator.go [hunk 1]
```diff
@@ -142,13 +142,30 @@ func parseExpression(e interface{}) (interface{}, error) {
 	return nil, fmt.Errorf("not supported expr: %v", el)
 }
 
+func inferStringValue(expr string) interface{} {
+	ret, err := strconv.Atoi(expr)
```
Comment (tonyyang-svail) [inline sql/ir_generator.go]: Maybe change to more concised statements
```go
if ret, err := strconv.Atoi(expr); err == nil {
    return ret
}
```

File: sql/ir_generator.go [hunk 2]
```diff
@@ -142,13 +142,30 @@ func parseExpression(e interface{}) (interface{}, error) {
...
+	if err == nil {
+		return ret
+	}
+	retFloat, err := strconv.ParseFloat(expr, 32)
+	if err == nil {
+		// NOTE: always use float32 for attributes, we may never use a
```
Comment (tonyyang-svail) [inline sql/ir_generator.go]: `NOTE` => `NOTE(typhoonzero)`. It is better to indicate the owner of the note.

File: sql/ir_generator.go [hunk 3]
```diff
@@ -142,13 +142,30 @@ func parseExpression(e interface{}) (interface{}, error) {
...
+}
+
 func parseFeatureColumn(el *exprlist) (codegen.FeatureColumn, error) {
 	head := (*el)[0].val
 	if head == "" {
 		return nil, fmt.Errorf("column description expects format like NUMERIC(key) etc, got %v", el)
 	}
 
 	switch strings.ToUpper(head) {
+	// TODO(typhoonzero): support FieldMeta configurations in column clause
```
Comment (tonyyang-svail) [inline sql/ir_generator.go]: I don't quite understand. How do we support FieldMeta configurations in column clause?

File: sql/ir_generator.go (hunk 2 above)
Comment (typhoonzero) [reply sql/ir_generator.go]: `golint` seems is not checking `// NOTE` comments, so I just removed `NOTE` here.

File: sql/ir_generator.go (hunk 3 above)
Comment (typhoonzero) [reply sql/ir_generator.go]: Updated comment to:

```
// TODO(typhoonzero): support FieldMeta configurations (DENSE/SPARSE) in column clause, e.g.
// CATEGORY_ID(DENSE(...))
```

**Review by tonyyang-svail — DISMISSED** [review]

**Review by tonyyang-svail — APPROVED** [review]

**Pull request merged**